src/Microsoft.PowerShell.Commands.Utility/commands/utility/CsvCommands.cs

```diff
@@ -967,7 +970,7 @@ internal string ConvertPropertyNamesCSV(IList<string> propertyNames)
                             AppendStringWithEscapeAlways(_outputString, propertyName);
                             break;
                         case BaseCsvWritingCommand.QuoteKind.AsNeeded:
-                            if (propertyName.Contains(_delimiter))
+                            if (propertyName.IndexOfAny(_escapeCharacters) != -1)
```

Comment:
Can we use `public static unsafe int IndexOfAny(ref char searchStart, char value0, char value1, char value2, char value3, int length)` to avoid allocation of `_escapeCharacters`?

Reply:
@iSazonov That sounds awesome! However, I'll admit I'm a Typescript developer, not a C# developer. Is that a [dotnet private method call](https://source.dot.net/#System.Private.CoreLib/SpanHelpers.Char.cs,1235)? I'm open to suggestions

Reply:
I see public https://source.dot.net/#System.Private.CoreLib/SpanHelpers.Char.cs,985

Reply:
I couldn't find any way to reference that method - I couldn't find any public references to it in github/code, and even tried converting the string to a [Span](https://docs.microsoft.com/en-us/dotnet/api/system.memoryextensions.indexofany?view=net-6.0#System_MemoryExtensions_IndexOfAny__1_System_Span___0____0___0___0_) but couldn't get it to work. It certainly seems preferable if you can show me how to use it / commit yourself.

FWIW I do see a [number of references](https://github.com/search?p=2&q=org%3APowerShell+indexofany&type=Code) to `string.IndexOfAny()` in this repo, but they aren't in this kind of inner loop situation where performance is a concern.

Reply:
Public is here https://source.dot.net/#System.Private.CoreLib/MemoryExtensions.cs,692

Reply:
Okay, [updated accordingly](https://github.com/PowerShell/PowerShell/pull/15765/commits/4832f43b91128e700b131a969c052e800ef6549f). I had to take out `\r` detection but I don't think that will matter since that character should always be followed by `\n`.

Reply:
> I had to take out `\r` detection

Why?